lib/dal/src/attribute/prototype/debug.rs

```diff
@@ -0,0 +1,207 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use thiserror::Error;
+
+use crate::attribute::prototype::{
+    argument::{
+        static_value::StaticArgumentValue,
+        value_source::{ValueSource, ValueSourceError},
+        AttributePrototypeArgument, AttributePrototypeArgumentError,
+    },
+    AttributePrototypeError,
+};
+use crate::attribute::value::{AttributeValueError, ValueIsFor};
+
+use crate::prop::{PropError, PropPath};
+
+use crate::func::execution::{FuncExecution, FuncExecutionError};
+use crate::socket::input::InputSocketError;
+use crate::socket::output::OutputSocketError;
+use crate::workspace_snapshot::node_weight::NodeWeightError;
+use crate::workspace_snapshot::WorkspaceSnapshotError;
+use crate::{
+    AttributePrototype, AttributePrototypeId, AttributeValue, AttributeValueId, Component,
+    ComponentError, DalContext, Func, FuncError, FuncId, InputSocket, Prop,
+};
+use serde_json::Value;
+use telemetry::prelude::*;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct AttributePrototypeDebugView {
+    pub func_id: FuncId,
+    pub func_execution: Option<FuncExecution>,
+    pub prototype_id: AttributePrototypeId,
+    pub func_name: String,
+    pub func_args: HashMap<String, Vec<serde_json::Value>>,
+    pub arg_sources: HashMap<String, Option<String>>,
+    pub all_attribute_values: Vec<AttributeValueId>,
+}
+
+type AttributePrototypeDebugViewResult<T> = Result<T, AttributePrototypeDebugViewError>;
+
+#[remain::sorted]
+#[derive(Error, Debug)]
+pub enum AttributePrototypeDebugViewError {
+    #[error(transparent)]
+    AttributePrototypeArgumentError(#[from] AttributePrototypeArgumentError),
+    #[error(transparent)]
+    AttributePrototypeError(#[from] AttributePrototypeError),
+    #[error(transparent)]
+    AttributeValue(#[from] AttributeValueError),
+    #[error(transparent)]
+    ComponentError(#[from] ComponentError),
+    #[error(transparent)]
+    Func(#[from] FuncError),
+    #[error(transparent)]
+    FuncExecution(#[from] FuncExecutionError),
+    #[error(transparent)]
+    InputSocketError(#[from] InputSocketError),
+    #[error(transparent)]
+    NodeWeightError(#[from] NodeWeightError),
+    #[error(transparent)]
+    OutputSocketError(#[from] OutputSocketError),
+    #[error(transparent)]
+    PropError(#[from] PropError),
+    #[error(transparent)]
+    ValueSourceError(#[from] ValueSourceError),
+    #[error(transparent)]
+    WorkspaceSnapshotError(#[from] WorkspaceSnapshotError),
+}
```

Comment:
I would not use `transparent` here since we do not want to forward the underlying error. We should probably add shapes like the following:

```rust
#[error("name of error lower case with spaces error: {0}")]
```

Relevant doc: https://docs.rs/thiserror/latest/thiserror/

Reply:
@vbustamante something for us to talk through so I can understand better!